src/rules/readme-lists-correct-yarn-version.ts

```diff
@@ -0,0 +1,32 @@
+import { buildRule } from './helpers';
+import { RuleName } from './types';
+
+export default buildRule({
+  name: RuleName.ReadmeListsCorrectYarnVersion,
+  description:
+    'Does the README conform by recommending the correct Yarn version to install?',
+  dependencies: [RuleName.RequireReadme, RuleName.AllYarnModernFilesConform],
+  execute: async ({ template, project, pass, fail }) => {
+    const entryPath = 'README.md';
+
+    const fileContentInTemplate = await template.fs.readFile(entryPath);
+    const fileContentInProject = await project.fs.readFile(entryPath);
+
+    const match = fileContentInTemplate.match(
+      /Install \[Yarn [^[\]]+\]\([^()]+\)/u,
+    );
+    if (!match?.[0]) {
+      throw new Error(
```

Comment:
My thought was a rule should only fail if the project does not meet whatever condition is being checked. In this case, we can't even run the rule, because the thing we're using as a point of comparison doesn't exist in the module template. I feel like this is a bug in the rule and not a deficiency in the project itself, so throwing an error seemed more appropriate so that it stood out and hopefully we could be notified of the problem and fix the bug.

That said, you are totally right that throwing an error would prevent other rules from being run, which doesn't seem ideal. I would probably need to fix this in `execute-rules.ts`.

Printing a warning (and returning a passing result) seems like a good solution for now, and then I can follow this up with another PR to properly address this. I've made this change here: [ae3da2c](https://github.com/MetaMask/module-lint/pull/39/commits/ae3da2c2a6fe457ad299c11a6a4ce3c7093a1c66)

Reply:
Sorry — reverted in [1b404db](https://github.com/MetaMask/module-lint/pull/39/commits/1b404dbb0a1e0c4ce64ce4f811d91955ccf4d5da) and [d2afd04](https://github.com/MetaMask/module-lint/pull/39/commits/d2afd043fa867da11048ef5d14c1c8358b3db623). For now, I think throwing is more accurate than printing a warning and then passing, which could be confusing without context as to why that's happening. I will still make a ticket to fix `execute-rules.ts`, however.